packages/holocron-module-register-webpack-plugin/HolocronModuleRegisterPlugin.js

```diff
@@ -31,14 +30,12 @@ HolocronModuleRegisterPlugin.prototype.apply = function apply(compiler) {
         chunk.files
           .filter(ModuleFilenameHelpers.matchObject.bind(undefined, options))
           .forEach((file) => {
-            // eslint-disable-next-line no-param-reassign
-            compilation.assets[file] = new ConcatSource(
-              '(function() {',
-              '\n',
-              compilation.assets[file],
-              '\n',
-              `Holocron.registerModule("${moduleName}", ${holocronModuleName});})();`
-            );
+            const source = compilation.assets[file];
+            // descend into the source and inject the registration within the iife
+            // The last two symbols are always the closing of the iife, then a `;`
+            // Therefore, insert the registration immediately before the iife closes
+            // eslint-disable-next-line no-underscore-dangle
+            source._source._children.splice(source._source._children.length - 2, 0, `;Holocron.registerModule("${moduleName}", ${holocronModuleName});`);
```

Comment:
I might be missing something but I don't see anything breaking in these changes.

Reply:
<img width="1548" alt="image" src="https://github.com/americanexpress/holocron/assets/2575450/fbd57f34-6a2c-44db-9db5-7b8715e2fa0e">
It breaks in webpack v4 because the structure of the `source` object has changed.

Reply:
```js
if(source?._source?._children) {
 // webpack 5 logic
} else {
 // webpack 4 logic
}
```

Reply:
I think we could also find another way to determine the version of webpack being used and alter logic based off that

Reply:
I considered this, but then this package would somehow need webpack 4 and 5 installed. And relying on peer-deps to get those in place could lead to problems with `--use-legacy-peerdeps`

Reply:
After discussion , independent versioning is going to be used